agent/consul/acl.go

```diff
@@ -192,7 +192,11 @@ func NewACLResolver(config *ACLResolverConfig) (*ACLResolver, error) {
 	}
 
 	if config.Logger == nil {
-		config.Logger = log.New(os.Stderr, "", log.LstdFlags)
+		consulLogger := hclog.New(&hclog.LoggerOptions{})
```

Comment:
This should probably explicitly set the Output at least during this massive transition so everything is crystal clear, even if Stderr is the default.

Possibly for non-test situations like this it might be useful to drop a helper function in `github.com/hashicorp/consul/lib` to roughly initialize the default hclog equivalent of `log.New(os.Stderr, "", log.LstdFlags)`. Though part of me is also super curious if practically speaking this ACLResolverConfig.Logger field is amusingly only ever `nil` during tests...